core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java

```diff
@@ -462,6 +465,33 @@ public void simpleStats() throws Exception {
         assertThat(stats.getTotal().getIndexing(), nullValue());
         assertThat(stats.getTotal().getGet(), nullValue());
         assertThat(stats.getTotal().getSearch(), nullValue());
+
+        // index failed
+        try {
+            client().prepareIndex("test1", "type1", Integer.toString(1)).setSource("field", "value").setVersion(1)
+                    .setVersionType(VersionType.EXTERNAL).execute().actionGet();
+        } catch (VersionConflictEngineException e) {
+            //expected
+        }
+        try {
+            client().prepareIndex("test1", "type2", Integer.toString(1)).setSource("field", "value").setVersion(1)
+                    .setVersionType(VersionType.EXTERNAL).execute().actionGet();
+        } catch (VersionConflictEngineException e) {
+            //expected
+        }
+        try {
+            client().prepareIndex("test2", "type", Integer.toString(1)).setSource("field", "value").setVersion(1)
+                    .setVersionType(VersionType.EXTERNAL).execute().actionGet();
+        } catch (VersionConflictEngineException e) {
+            //expected
+        }
+        refresh();
+
+        stats = client().admin().indices().prepareStats().setTypes("type1").execute().actionGet();
```

Comment:
> I dont think we need to check for each type, but I can add it if you think its needed.

I don't think its wrong to check both types. Its nice to prove that they count independently I guess. I certainly don't think its wrong to test it like you are. I just thought that you'd need type2 because you assert things about it a couple lines below.